internal/app/init/pkg/system/system.go

```diff
@@ -23,12 +31,12 @@ var once sync.Once
 type Service interface {
 	// ID is the service id.
 	ID(*userdata.UserData) string
-	// PreFunc is invoked before a command is executed.
-	PreFunc(*userdata.UserData) error
-	// Start
-	Start(*userdata.UserData) error
-	// PostFunc is invoked after a command is executed.
-	PostFunc(*userdata.UserData) error
+	// PreFunc is invoked before a runner is created
+	PreFunc(context.Context, *userdata.UserData) error
```

Comment:
context added to allow methods to be cancellable when init is shutting down. this might be overkill, seems we don't use it

Reply:
decided to remove it